int8 x
int8 y
float64 dist
string voice